src/app/containers/RedeemContainer.js

```diff
@@ -1,7 +1,179 @@
+    const [couponCode, setCouponCode] = useState();
+    const [validationError, setValidationError] = useState();
+
+    const handleCouponCodeChange = useCallback(({ target: { value } }) => {
+        setCouponCode(value);
+    }, []);
+
+    const validateCoupon = useCallback(
+        async (e) => {
+            e.preventDefault();
+
+            try {
+                const {
+                    Plans: [{ ID, Currency, Cycle, Quantity }]
+                } = await api(queryCheckVerificationCode(couponCode, TOKEN_TYPES.COUPON, CLIENT_TYPE));
+
+                const response = await api(
+                    checkSubscription({
+                        CouponCode: couponCode,
+                        Currency,
+                        Cycle,
+                        PlanIDs: {
+                            [ID]: Quantity
+                        }
+                    })
+                );
+
+                history.push('/signup', response);
+            } catch (e) {
+                if (!e.data) {
+                    throw e;
+                }
+                switch (e.data.Error) {
+                    case 'Token invalid':
+                        setValidationError('Invalid coupon. Please try again.');
+                        break;
+                    default:
+                        setValidationError(e.data.Error);
+                }
+            }
+        },
+        [couponCode]
+    );
+
+    return (
+        <div>
+            <div className="flex flex-items-center flex-justify-start onmobile-flex-column pt1 pb1 bg-global-grey">
+                <div className="flex-item-fluid">
+                    <div className="mw80 center">
+                        <Href className="color-white nodecoration" url="https://protonvpn.com" target="_self">
+                            <Icon name="caret" color="white" rotate={90} />
+                            {c('Link').t`Return to protonvpn.com`}
+                        </Href>
+                    </div>
+                </div>
+            </div>
+            <div className="redeem-header-wrapper pt1 pb1 bg-global-grey">
+                <div className="mw80 center">
+                    <header className="header flex flex-items-center flex-nowrap reset4print">
+                        <VpnLogo url="/account" />
+                    </header>
+                </div>
+            </div>
+            <div className="redeem-heading-bg pt1">
+                <div className="mw80 center">
+                    <Title className="redeem-heading-title mt2 mb1-5 color-white bold">
+                        {c('Title').t`Redeem your coupon and start protecting your online privacy`}
+                    </Title>
+                    <div className="flex">
+                        <form className="redeem-form w100 col flex flex-column pb2" onSubmit={validateCoupon}>
+                            <Input
+                                className="redeem-coupon-code-input"
```

Comment:
Use `error` prop instead of:
``` jsx
{!!validationError && (
                                <ErrorZone className="bg-global-warning color-white aligncenter p0-5">
                                    {c('Error').jt`${validationError}`}
                                </ErrorZone>
                            )}
```